### PR TITLE
Linkage Monitor to read content of a BOM

### LINKED ISSUE #1919: Linkage Monitor fails to detect linkage errors in gax-java
Linkage Monitor fails to detect linkage errors in gax-java.

Linkage Monitor has capability to replace a BOM in the libraries BOM. However, we no longer specify gax-bom in it. Therefore current it does not detect anything in gax-java's repository.

This bug is only applicable for gax-java repository, which relies on gax-java BOM to tell locally installed artifacts. Other Maven projects in Yoshi repositories, which produces pom.xml files in project directories, are not affected.

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #1919

### CODE REVIEW

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 1]
```diff
@@ -188,14 +190,28 @@ private static void printUsageAndDie() {
       try {
         ModelBuildingResult modelBuildingResult = modelBuilder.build(modelRequest);
         Model model = modelBuildingResult.getEffectiveModel();
-        artifactToVersion.put(model.getGroupId() + ":" + model.getArtifactId(), model.getVersion());
+        String versionlessCoordinates = model.getGroupId() + ":" + model.getArtifactId();
+        artifactToVersion.put(versionlessCoordinates, model.getVersion());
+        logger.info("Found local artifact: " + model);
```
Comment (elharo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java:195]: info message not needed

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 2]
```diff
@@ -188,14 +190,28 @@ private static void printUsageAndDie() {
...
+        DependencyManagement dependencyManagement = model.getDependencyManagement();
+        if ("pom".equals(model.getPackaging()) && dependencyManagement != null) {
+          // Read the content of a BOM.
+          for (org.apache.maven.model.Dependency dependency :
+              dependencyManagement.getDependencies()) {
+            String managedDependencyVersionlessCoordinates =
+                dependency.getGroupId() + ":" + dependency.getArtifactId();
+            artifactToVersion.put(managedDependencyVersionlessCoordinates, dependency.getVersion());
+            logger.info("Found local artifact in the BOM: " + dependency);
```
Comment (elharo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java:204]: unnecessary

File: linkage-monitor/src/test/resources/testproject/gax-bom/pom.xml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  <scm>
+    <connection>scm:git:https://github.com/googleapis/gax-java.git</connection>
+    <url>https://github.com/googleapis/gax-java</url>
+  </scm>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.google.api</groupId>
+        <artifactId>gax</artifactId>
+        <version>1.60.2-SNAPSHOT</version><!-- {x-version-update:gax:current} -->
```
Comment (elharo) [inline linkage-monitor/src/test/resources/testproject/gax-bom/pom.xml:36]: we shouldn't need to version update the test data

File: linkage-monitor/src/test/resources/testproject/gax-bom/pom.xml [hunk 4]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.api</groupId>
+  <artifactId>gax-bom</artifactId>
+  <version>1.60.2-SNAPSHOT</version><!-- {x-version-update:gax-bom:current} -->
+  <packaging>pom</packaging>
+  <name>GAX (Google Api eXtensions) for Java</name>
```
Comment (elharo) [inline linkage-monitor/src/test/resources/testproject/gax-bom/pom.xml:8]: Most of this metadata isn't needed for the test

**Review by suztomo — COMMENTED** [review]
@elharo PTAL.

**Review by elharo — APPROVED** [review]

**Pull request merged**